src/compiler/diagnosticMessages.json

```diff
@@ -671,6 +671,10 @@
         "category": "Error",
         "code": 1215
     },
+    "Identifier expected. '__esModule' is reserved as a marker for transpiled ES2015 module.": {

```

Comment:
I like the new message. But Identifier expected should be there I think ( similar to how we report similar error for reserved word)